source/common/http/http1/codec_impl.cc

```diff
@@ -843,6 +848,11 @@ void ConnectionImpl::dispatchBufferedBody() {
 }
 
 void ConnectionImpl::onChunkHeader(bool is_final_chunk) {
+  StreamInfo::BytesMeterer* bytes_meterer = getBytesMeterer();
+  // Count overhead of chunk encoding per chunk.
+  if (bytes_meterer) {
+    bytes_meterer->addBodyBytesReceived(is_final_chunk ? 3 : 4);
```

Comment:
If not the final chunk the length is >= 4, you need to include the bytes used for the chunk length

e.g. `20\r\n<20 chars of data....>\r\n` need to include 2 bytes for the 20.